Branch-SDK/Branch-SDK/BNCStrongMatchHelper.m

```diff
@@ -95,69 +155,121 @@ + (NSURL *)getUrlForCookieBasedMatchingWithBranchKey:(NSString *)branchKey redir
-            [self.secondWindow makeKeyWindow];
-            
-            // Give enough time for Safari to load the request (optimized for 3G)
-            dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(3 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
-                [keyWindow makeKeyWindow];
-                
-                // Remove the window and release it's strong reference. This is important to ensure that
-                // applications using view controller based status bar appearance are restored.
-                [self.secondWindow removeFromSuperview];
-                self.secondWindow = nil;
-                
-                [BNCPreferenceHelper preferenceHelper].lastStrongMatchDate = [NSDate date];
+
+            if (![self willLoadViewControllerWithURL:strongMatchUrl]) {
+                self.shouldDelayInstallRequest = NO;
                 self.requestInProgress = NO;
-            });
+                return;
+            }
+
+            // Give enough time for Safari to load the request (optimized for 3G)
+            dispatch_after(
+                dispatch_time(DISPATCH_TIME_NOW, (int64_t)(3 * NSEC_PER_SEC)),
+                dispatch_get_main_queue(),
+                ^{ [self unloadViewController]; }
+            );
         });
     }
-    else {
-        self.requestInProgress = NO;
+}
+
+- (BOOL) subclass:(Class)subclass selector:(SEL)selector {
+    Class templateClass = objc_getClass("BNCMatchViewController");
+    if (!templateClass) return NO;
+
+    Method method = class_getInstanceMethod(templateClass, selector);
+    if (!method) return NO;
+
+    const char * typeEncoding = method_getTypeEncoding(method);
+    if (!typeEncoding) return NO;
+
+    IMP implementation = class_getMethodImplementation(templateClass, selector);
+    if (!implementation) return NO;
+
+    class_addMethod(subclass, selector, implementation, typeEncoding);
+    return YES;
+}
+
+- (BOOL) willLoadViewControllerWithURL:(NSURL*)matchURL {
+    if (self.primaryWindow) return NO;
+
+    //  Dynamically subclass the SFSafariViewController if available.
+    //  This allows us to compile and link to an app that doesn't
+    //  include SafariServices, but is also able to compile and link
+    //  when it is.
+
+    Class SFSafariViewControllerClass = NSClassFromString(@"SFSafariViewController");
+    if (!SFSafariViewControllerClass) return NO;
+
+    Class BNCMatchViewControllerSubclass = NSClassFromString(@"BNCMatchViewController_Safari");
+    if (!BNCMatchViewControllerSubclass) {
+        //  The class isn't registered.  Create it:
+
+        BNCMatchViewControllerSubclass =
+            objc_allocateClassPair(SFSafariViewControllerClass, "BNCMatchViewController_Safari", 0);
+        if (!BNCMatchViewControllerSubclass) return NO;
+
+        BOOL fail = NO;
+        fail |= ![self subclass:BNCMatchViewControllerSubclass selector:@selector(becomeFirstResponder)];
```

Comment:
Why are you hooking these up? The methods don't do anything? Maybe this PR isn't ready for review?